### DISCUSSION TITLE
Documentation missed the nginx configuration which serves shlink in a subpath

### DISCUSSION THREAD

**Discussion by Grief** [shlinkio/discussions#82] [Help wanted]
The documentation contains the snippet with the configuration for nginx (https://shlink.io/documentation/classic-web-server/#nginx), but it assumes that shlink is located in a root path. It's unclear how to configure nginx to server shlink in a sub-path, e.g. http://example.com/shlink

**Comment by Grief** [comment]
I've tried something like
```
location /shlink {
    alias /srv/shlink/public;
    index index.php;

    location ~ \.php$ {
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $request_filename;
        fastcgi_pass php-8;
    }
}

```
but it leads to 404 be shown:
![image](https://user-images.githubusercontent.com/7239315/152806596-1e5987e6-d03a-4516-a1c9-8a8b5bc04f08.png)

**Reply by acelaya** [reply]
Have you informed Shlink about the base path with the `BASE_PATH` env var? https://shlink.io/documentation/environment-variables/#general

**Reply by Grief** [reply]
Yes, I did. I tried both '/shlink' and 'shlink' variants with no luck
```
cat config/params/generated_config.php|grep BASE_PATH 
    'BASE_PATH' => '/shlink',
```

**Reply by Grief** [reply]
@acelaya could it be a bug in shlink? I see that 'visits' table is filled by 'base-url' types, not 'regular_404', but I still see 404
![image](https://user-images.githubusercontent.com/7239315/152969488-3f313190-5309-4af2-9013-cd5f848f87c8.png)

**Reply by acelaya** [reply]
And what's not working as expected there? If it's detecting it as base_url it means it works.

**Reply by Grief** [reply]
Web interface is not working, showing 404 (screenshot is above)

**Reply by Grief** [reply]
Oh. My. God. I am stupid. this is not a web interface, but rest service only. web-client must be installed separately. I'll try to install it

**Reply by Grief** [reply]
Okay, it still doesn't work. First of all, it provides two-level links, like example.com/shlink/link/XXX, secondly these links lead to nginx 404.

**Reply by acelaya** [reply]
I think the problem is on your nginx config. Maybe you can try something like this https://serverfault.com/a/778336

**Reply by Grief** [reply]
I tried 'something like' that and it didn't work at all:
```
location ^~ /link {
    root /srv/shlink/public;
    index index.php;
    try_files $uri $uri/ /link/index.php;

    location ~ \.php$ {
        try_files $uri =404;
        include params/fastcgi.conf;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_pass php-8;
    }
}
``` 
I get nginx 404 for every request

I doubt that the configuration you are referring to is applicable is that case because when you use 'root' directive inside the 'location', the filesystem path is joined from root AND location. But the location must be skipped in this case, so 'alias' directive should be used, I guess. That didn't work too though.
